brightwind/analyse/plot.py

```diff
@@ -169,22 +169,185 @@ def plot_monthly_means(data, coverage=None, ylbl=''):
+                                        2) List of str or Hex or Rgb: the number of colors provided needs to be
+                                           at least equal to the number of columns in the y input.
+                                        3) None: the default matplotlib color list will be used for plotting.
+    :type line_colors:              str or list or tuple or None
+    :param subplot_title:           Title show on top of the subplot. Default is None.
+    :type subplot_title:            str or None
+    :param legend:                  Boolean to choose if legend is shown. Default is True.
+    :type legend:                   Bool
+    :param ax:                      Subplot axes to which assign the subplot to in a plot. If None then a single plot is
+                                    generated
+    :type ax:                       matplotlib.axes._subplots.AxesSubplot or None
+    :return:                        A timeseries subplot
+    :rtype:                         matplotlib.axes._subplots.AxesSubplot
+
+     **Example usage**
+    ::
+        import brightwind as bw
+        import matplotlib.pyplot as plt
+        data = bw.load_csv(bw.demo_datasets.demo_data)
+        wspd_cols = ['Spd80mN', 'Spd80mS', 'Spd60mN', 'Spd60mS', 'Spd40mN', 'Spd40mS']
+
+        # To plot only one subplot in a figure and set different marker types for each line
+        fig, axes = plt.subplots(1, 1)
+        bw.analyse.plot._timeseries_subplot(data.index, data[wspd_cols],
+                                            line_marker_types=['.', 'o', 'v', '^', '<', None], ax=axes)
+
+        # To plot multiple subplots in a figure without legend and with x and y labels and assigning subplot_title
+        fig, axes = plt.subplots(2, 1)
+        bw.analyse.plot._timeseries_subplot(data.index, data[wspd_cols], ax=axes[0], legend=False,
+                                         x_label=None, y_label='Spd80mS', subplot_title='Speed [m/s]')
+        bw.analyse.plot._timeseries_subplot(data.index, data.Dir78mS, ax=axes[1], legend=False,
+                                         x_label='Time', y_label='Dir78mS', subplot_title='Direction [deg]')
+
+        # To plot multiple timeseries with different x values/length in the same subplot,
+        # only for 1st timeseries set marker type and color different than default, set legend for all timeseries
+        fig, axes = plt.subplots(1, 1)
+        ts_plot = bw.analyse.plot._timeseries_subplot(data['2016-02-10':'2016-03-10'].index,
+                                                      data['2016-02-10':'2016-03-10'].Spd60mS, line_marker_types='.',
+                                                      line_colors=bw.analyse.plot.COLOR_PALETTE.tertiary, ax=axes)
+        ts_plot = bw.analyse.plot._timeseries_subplot(data.index, data[['Spd80mS','Spd60mN']], ax=axes)
+        ts_plot.axes.legend(['Spd60mS', 'Spd80mS','Spd60mN',])
```

Comment:
@BiancaMorandi I'm going through the examples and I'm not sure what this line does?
`ts_plot.axes.legend(['Spd60mS', 'Spd80mS','Spd60mN',])`

Reply:
this is used to customise the legend after plotting. I took that out now as legend is ok also without this.